src/main/java/com/google/longrunning/PagedResponseWrappers.java

```diff
@@ -47,15 +48,116 @@
-        PagedListDescriptor<ListOperationsRequest, ListOperationsResponse, Operation>
-            pageDescriptor,
-        ListOperationsRequest request,
-        CallContext context) {
-      super(callable, pageDescriptor, request, context);
+      extends AbstractPagedListResponse<ListOperationsRequest, ListOperationsResponse, Operation> {
+
+    private final ListOperationsPage page;
+
+    public static ListOperationsPagedResponse callApiAndCreate(
+        PageContext<ListOperationsRequest, ListOperationsResponse, Operation> context) {
+      ListOperationsPage page = ListOperationsPage.callApiAndCreate(context);
+      return new ListOperationsPagedResponse(page);
+    }
+
+    private ListOperationsPagedResponse(ListOperationsPage page) {
+      this.page = page;
+    }
+
+    @Override
+    public ListOperationsPage getPage() {
+      return page;
+    }
+
+    @Override
+    public Iterable<ListOperationsPage> iteratePages() {
+      return page.iteratePages();
+    }
+
+    @Override
+    public ListOperationsFixedSizeCollection expandToFixedSizeCollection(int collectionSize) {
+      return ListOperationsFixedSizeCollection.expandPage(page, collectionSize);
+    }
+
+    @Override
+    public Iterable<ListOperationsFixedSizeCollection> iterateFixedSizeCollections(
+        final int collectionSize) {
+      return expandToFixedSizeCollection(collectionSize).iterateCollections();
+    }
+  }
+
+  public static class ListOperationsPage
+      extends AbstractPage<ListOperationsRequest, ListOperationsResponse, Operation> {
+
+    private static final PageFactory<
+            ListOperationsRequest, ListOperationsResponse, Operation, ListOperationsPage>
+        PAGE_FACTORY =
+            new PageFactory<
+                ListOperationsRequest, ListOperationsResponse, Operation, ListOperationsPage>() {
+              @Override
+              public ListOperationsPage createPage(
+                  PageContext<ListOperationsRequest, ListOperationsResponse, Operation> context,
+                  ListOperationsResponse response) {
+                return new ListOperationsPage(context, response);
+              }
+            };
+
+    public static ListOperationsPage callApiAndCreate(
+        PageContext<ListOperationsRequest, ListOperationsResponse, Operation> context) {
+      return callApiAndCreate(PAGE_FACTORY, context);
+    }
+
+    private ListOperationsPage(
+        PageContext<ListOperationsRequest, ListOperationsResponse, Operation> context,
+        ListOperationsResponse response) {
+      super(context, response);
+    }
+
+    @Override
+    public ListOperationsPage getNextPage() {
+      return getNextPage(PAGE_FACTORY);
+    }
+
+    @Override
+    public ListOperationsPage getNextPage(int pageSize) {
+      return getNextPage(PAGE_FACTORY, pageSize);
+    }
+
+    private Iterable<ListOperationsPage> iteratePages() {
+      return iterate(PAGE_FACTORY, this);
```

Comment:
I think that either one of the following should be done:

1. Provide `PAGE_FACTORY` to the `AbstractPage` constructor so it has a reference to it, and thus the concrete classes don't have to pass the page factory up, and these functions can be pure `super.X` calls
2. Add a "factory method" to `AbstractPage` to create an instance of the concrete page, which is implemented in the concrete classes, and avoid the page factory altogether:

```
public class ListOperationsPage
    extends AbstractPage<ListOperationsRequest, ListOperationsResponse, Operation> {
  @Override
  public ListOperationsPage createNewInstance(
      PageContext<ListOperationsRequest, ListOperationsResponse, Operation> context,
      ListOperationsResponse response) {
    return new ListOperationsPage(context, response);
  }
}
```